### ISSUE TITLE
[3.x] 2D Particle hierarchical culling not working correctly

### ISSUE THREAD

**Issue by lawnjelly** [godotengine/godot#80086]
### Godot version

3.6 (after #68738 merged)

### System information

All

### Issue description

2D particles are being culled when the parent object leaves the frame, when hierarchical culling is enabled.

https://github.com/godotengine/godot/assets/21999379/1ec3575d-0711-4bea-b152-b673cd3249ca

### Steps to reproduce

Create 2D particle system and move off screen with hierarchical culling enabled. Note that the particles disappear as soon as the origin moves off screen, rather than still showing the particles that have been emitted.

### Minimal reproduction project

[particle_hierarchical_culling.zip](https://github.com/godotengine/godot/files/12215342/particle_hierarchical_culling.zip)

## Discussion
This is easy enough to fix in the meantime by turning off hierarchical culling (`rendering/2D/options/culling_mode` set to `Item` instead of `Node`).

I'm just fixing this now. The problem is the same as with skeletons - the bounding rect is changed by updates from the scene tree, and the local bound is never updated, so everything is culled from the parent object.

In the legacy item culling code, the bound is deferred updated when `get_rect()` is called on the item during the render traversal through the tree.

There's a number of ways to solve this, but I'll probably use something similar to skeletons, and trigger an update of the local bounds when the multimesh is changed (at least in the case of `CPUParticles2D`, haven't got to `GPUparticles2D` yet).
